store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8156,6 +8160,39 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         return result;
     }
 
+    /**
+     * It modifies contact's first and full name with display name
+     * @param octxt
+     * @param listOfContactsToBeModified This list consist list of Map, InternetAddress as key and Pair of ContactId and 
+     * DisplayName as Value
+     */
+    public void modifyAutoContact(OperationContext octxt, List<Map<InternetAddress, Pair<Integer, String>>> listOfContactsToBeModified)
+            throws IOException {
+        String email = null;
+        String locale = octxt != null && octxt.getAuthenticatedUser() != null ? octxt.getAuthenticatedUser().getPrefLocale() : null;
+        boolean nameFormatLastFirst = false;
+        if (locale != null && locale.equals("ja")) {
```

Comment:
nameFormatLastFirst can be true for a list of countries including japan. 
Can we make it a list and check for contains there?

Reply:
I am informed/asked to do for Japanese only since in Japanese LastName come first and also I was asked to refer createAutoContact() where it is handled  already in that and working fine. Please refer: https://github.com/Zimbra/zm-mailbox/pull/1234#pullrequestreview-913846597
Also, I am not aware of other countries where this LastName come first, could you be able to provide me the same so that I can check for those countries, if necessary.